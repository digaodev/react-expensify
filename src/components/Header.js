import React from 'react';

import { connect } from 'react-redux';

import { NavLink } from 'react-router-dom';

import { startLogout } from '../actions/auth';

const Header = ({ handleLogout }) => {
  return (
    <header>
      <h1>Expensify</h1>
      <nav>
        <NavLink activeClassName="is-active" to="/dashboard">
          Dashboard
        </NavLink>
        <NavLink activeClassName="is-active" to="/add">
          Add Expense
        </NavLink>
        <button type="button" onClick={handleLogout}>
          Logout
        </button>
      </nav>
    </header>
  );
};

const mapDispatchToProps = dispatch => ({
  handleLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
