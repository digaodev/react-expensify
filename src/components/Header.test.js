import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from './Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);

  expect(wrapper.find('h1').text()).toBe('Expensify');
  expect(toJson(wrapper)).toMatchSnapshot();
});
