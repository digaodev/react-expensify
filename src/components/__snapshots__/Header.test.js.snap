// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/add"
    >
      Add Expense
    </NavLink>
  </nav>
</header>
`;
